book/browsersandweb.md

```diff
@@ -5,71 +5,52 @@ next: http
-computing; declarative document models for describing UIs; hyperlinks; and the
-User Agent concept.
-
-Where did these concepts come from? Were they invented all at once for the web?
-And aren't they kind of simple or even obvious? The answers are: many places,
-no, and only in hindsight. These concepts are the the result of a large amount
-of experimentation and research reaching back to nearly the beginning of
-computing. The web itself only appeared decades later, in 1989, and stood on the
-shoulders of that work to put those concepts together in just the right way. And
-as with many technological breakthroughs, it also appeared at just the right
-time---in particular around the point at which the enabling technologies of
-computer displays, powerful UI-building libraries, and networking passed
-critical thresholds of maturity.
-
-At the core of the web-based computing vision is the browser. The browser is the
-User Agent[^useragent], the _mediator_ of web interactions and _enforcer_ of its
-rules. Not only that, the browser is the _implementer_ of all of the ways
-information is explored---in particular through algorithms to make the "black
-box" of the declarative document model as expressive and rich as possible, while
-at the same time adhering strictly to the web's decentralized architecture and
-commitment to openness. And of course, the web would be no fun if websites were
-not fast to load and smooth to use, so the browser must be hyper-efficient as
-well.
-
-[^useragent]: The User Agent is the concept that the computer, or trusted
-software on the computer, is one’s trusted assistant and advocate; browsers are
-a kind of User Agent.
-
-Figuring out how to meet all these lofty goals with real code is certainly
-difficult, but _also_ groundbreaking, endlessly fascinating even in the details,
-and often  just plain fun. That's what this book is about.
+    [web pages](https://en.wikipedia.org/wiki/Web_page) on the Internet.
+
+Nowadays, desktop applications are often built and delivered as _web
+apps_: websites[^website] used in similar ways to installed
+applications. And on mobile devices even native apps often use _web
+views_ that embed a browser render parts of the application UI.
+Perhaps in the future mobile devices will, like desktop computers,
+mostly a container for web apps.
+
+[^website]: You probably already know what a website is. This is one.
+    If not, [see here](https://en.wikipedia.org/wiki/Website).
+
+[^hybrid]: The fraction of these _hybrid apps_ that are web content is
+    also likely increasing over time.
+
+Yet this critical piece of software is built on a few simple, yet
+revolutionary, concepts that present a vision of the future of
+software and information: open, decentralized and safe computing;
+declarative UIs; hyperlinks; and user agents[^useragent]. This vision
+is neither simple nor obvious, even if it seems that way in hindsight:
```

Comment:
Here you changed the argument to emphasize the other technologies that made the web possible, but skipped the part about prior research into information systems. Was that on purpose?

Reply:
Yeah, I think I did not do this well. The real thought here was to keep the focus on browsers, whereas this paragraph changes focus to the web and concepts that aren't directly related to it. I tried swapping parts between this and the next paragraph, and that didn't really work, so I kind of ended up with this.

The idea of focusing on browsers still seems good but I think you can do a better job than I did.

Reply:
Ok. Revert this part or I'll try to patch in parts of your PR?

Reply:
You make a good point about the change of focus. Here is another take on resolving this issue. Better?

https://github.com/pavpanchekha/emberfox/pull/53